dspace-server-webapp/src/main/java/org/dspace/app/rest/CommunityRestController.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.UUID;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.dspace.app.rest.converter.BitstreamConverter;
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.CommunityRest;
+import org.dspace.app.rest.model.hateoas.BitstreamResource;
+import org.dspace.app.rest.repository.CommunityRestRepository;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Community;
+import org.dspace.content.service.CommunityService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ControllerUtils;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.hateoas.ResourceSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * This RestController takes care of the creation and deletion of Communities' nested objects
+ * This class will typically receive the UUID of a Community and it'll perform logic on its nested objects
+ */
+@RestController
+@RequestMapping("/api/" + CommunityRest.CATEGORY + "/" + CommunityRest.PLURAL_NAME)
+public class CommunityRestController {
```

Comment:
I would recommend renaming this Controller to be `CommunityLogoUploadController` or `CommunityLogoController`.  It's current name is misleading as this Controller does NOT implement the entire `/communities` endpoint...it ONLY implements the Logo upload feature.

Additionally, the `@RequestMapping` should be more specific to the `/logo` path....likely something like 

`@RequestMapping("/api/" + CommunityRest.CATEGORY + "/" + CommunityRest.PLURAL_NAME + "/" + REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID + "/logo")`

Reply:
NOTE: The ITs related to this Controller should also be renamed